diff --git a/dist/transport.js b/dist/transport.js
index 9093ba90e7a3d2e0a301f315bf9f5d908ba1d23f..80dbe7277f46d4dd1ff4fd0d3f8ab195d3a4ae2f 100644
--- a/dist/transport.js
+++ b/dist/transport.js
@@ -281,6 +281,9 @@ class Transport extends core.Transport {
                     } else {
                         throw new core.ActionCancelled();
                     }
+                } else if (response.message_enum === Messages.MessageType.MESSAGETYPE_FAILURE &&
+                    response.message.code === Types.FailureType.FAILURE_PINCANCELLED) {
+                    throw new core.ActionCancelled();
                 }
                 if (response.message_type === core.Events.FAILURE)
                     throw response;
@@ -311,7 +314,10 @@ class Transport extends core.Transport {
             try {
                 this.callInProgress = { main: undefined, debug: undefined };
                 const cancelMsg = new Messages.Cancel();
-                yield this.call(Messages.MessageType.MESSAGETYPE_CANCEL, cancelMsg);
+                yield this.call(Messages.MessageType.MESSAGETYPE_CANCEL, cancelMsg, {
+                    omitLock: this.userActionRequired,
+                    noWait: this.userActionRequired,
+                });
             }
             catch (e) {
                 console.error("Cancel Pending Error", e);
