diff --git a/dist/keepkey.js b/dist/keepkey.js
index 77e1f26f52b65c04978d264b2464e53c3b519c3f..8222208f1785c7e8fcd9f8ed76bad8a17e49c4bb 100644
--- a/dist/keepkey.js
+++ b/dist/keepkey.js
@@ -932,12 +932,14 @@ class KeepKeyHDWallet {
     // GetFeatures returns the features and other device information such as the version, label, and supported coins
     getFeatures(cached = false) {
         return __awaiter(this, void 0, void 0, function* () {
-            if (cached && this.featuresCache)
-                return this.featuresCache;
-            const features = new Messages.GetFeatures();
-            const event = yield this.transport.call(Messages.MessageType.MESSAGETYPE_GETFEATURES, features);
-            this.cacheFeatures(event.message);
-            return event.message;
+            return this.transport.lockDuring(() => __awaiter(this, void 0, void 0, function* () {
+                if (cached && this.featuresCache)
+                    return this.featuresCache;
+                const features = new Messages.GetFeatures();
+                const event = yield this.transport.call(Messages.MessageType.MESSAGETYPE_GETFEATURES, features);
+                this.cacheFeatures(event.message);
+                return event.message;
+            }));
         });
     }
     cacheFeatures(features) {
